src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/PackageRestoreConfiguredDataSource.cs

```diff
@@ -0,0 +1,82 @@
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks.Dataflow;
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+using NuGet.SolutionRestoreManager;
+
+using RestoreUpdate = Microsoft.VisualStudio.ProjectSystem.IProjectVersionedValue<Microsoft.VisualStudio.ProjectSystem.VS.NuGet.ProjectRestoreUpdate>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    /// <summary>
+    ///     Provides an implementation of <see cref="IPackageRestoreConfiguredDataSource"/> that combines evaluations results
+    ///     of <see cref="DotNetCliToolReference"/>, <see cref="ProjectReference"/> and build versions of <see cref="PackageReference"/> 
+    ///     into <see cref="ProjectRestoreUpdate"/>.
+    /// </summary>
+    [Export(typeof(IPackageRestoreConfiguredDataSource))]
+    internal partial class PackageRestoreConfiguredDataSource : ChainedProjectValueDataSourceBase<ProjectRestoreUpdate>, IPackageRestoreConfiguredDataSource
+    {
+        private static readonly ImmutableHashSet<string> s_rules = Empty.OrdinalIgnoreCaseStringSet
+                                                                        .Add(NuGetRestore.SchemaName)               // Evaluation
+                                                                        .Add(ProjectReference.SchemaName)           // Evaluation
+                                                                        .Add(PackageReference.SchemaName)           // Project Build
+                                                                        .Add(DotNetCliToolReference.SchemaName);    // Evaluation
+
+        private readonly ConfiguredProject _project;
+        private readonly IProjectSubscriptionService _projectSubscriptionService;
+
+        [ImportingConstructor]
+        public PackageRestoreConfiguredDataSource(ConfiguredProject project, IProjectSubscriptionService projectSubscriptionService)
+            : base(project.Services, synchronousDisposal: true, registerDataSource: false)
+        {
+            _project = project;
+            _projectSubscriptionService = projectSubscriptionService;
+        }
+
+        protected override UnconfiguredProject ContainingProject
+        {
+            get { return _project.UnconfiguredProject; }
+        }
+
+        protected override IDisposable LinkExternalInput(ITargetBlock<RestoreUpdate> targetBlock)
+        {
+            IProjectValueDataSource<IProjectSubscriptionUpdate> source = _projectSubscriptionService.JointRuleSource;
+
+            // Transform the changes from evaluation/design-time build -> restore data
+            DisposableValue<ISourceBlock<RestoreUpdate>> transformBlock = source.SourceBlock
+                                                                                .TransformWithNoDelta(update => update.Derive(u => CreateRestoreUpdate(u.CurrentState)),
```

Comment:
You cannot skip version only changes.  I guess....

Reply:
Why? I can ship version changes up until I start trying to coordinate the versions for operation progress or for the Unconfigured merge.

Reply:
Ah, of course not. I just realized why not - we're trying to coordinate these versions upstream and unless config publishes every version, they may never agree.